tests/vmi_configuration_test.go

```diff
@@ -566,58 +566,102 @@ var _ = Describe("[sig-compute]Configurations", decorators.SigCompute, func() {
-				secret := libsecret.New(secretWithSlicName, libsecret.DataBytes{"slic.bin": slicTable})
-				secret, err := virtClient.CoreV1().Secrets(testsuite.GetTestNamespace(vmi)).Create(context.Background(), secret, metav1.CreateOptions{})
+			for i := range numTables {
+				By("Creating secrets with the binary ACPI table")
+				ithSecretName := suffix(secretName, i)
+				ithVolumeName := suffix(volumeSecretName, i)
+				var secretRef *k8sv1.Secret
+				switch {
+				case hasSlic && !slicSet:
+					vmi.Spec.Domain.Firmware.ACPI.SlicNameRef = ithVolumeName
+					slicSet = true
+					secretRef = libsecret.New(ithSecretName, libsecret.DataBytes{
+						"slic.bin": slicData,
+					})
+				case hasMsdm && !msdmSet:
+					vmi.Spec.Domain.Firmware.ACPI.MsdmNameRef = ithVolumeName
+					msdmSet = true
+					secretRef = libsecret.New(ithSecretName, libsecret.DataBytes{
+						"msdm.bin": msdmData,
+					})
+				}
+				secret, err := virtClient.CoreV1().
+					Secrets(testsuite.GetTestNamespace(vmi)).
+					Create(context.Background(), secretRef, metav1.CreateOptions{})
 				Expect(err).ToNot(HaveOccurred())
 				Expect(secret).ToNot(BeNil())
 
 				By("Configuring the volume with the secret")
 				vmi.Spec.Volumes = append(vmi.Spec.Volumes, v1.Volume{
-					Name: volumeSlicSecretName,
+					Name: ithVolumeName,
 					VolumeSource: v1.VolumeSource{
 						Secret: &v1.SecretVolumeSource{
-							SecretName: secretWithSlicName,
+							SecretName: ithSecretName,
 						},
 					},
 				})
 
-				// The firmware needs to reference the volume name of slic secret
-				By("Configuring the firmware option with volume name that contains the secret")
-				vmi.Spec.Domain.Firmware = &v1.Firmware{
-					ACPI: &v1.ACPI{
-						SlicNameRef: volumeSlicSecretName,
-					},
-				}
-				vmi = libvmops.RunVMIAndExpectLaunch(vmi, 360)
-				Expect(console.LoginToAlpine(vmi)).To(Succeed())
+			}
 
-				By("Checking the guest ACPI SLIC table matches the one provided")
-				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+			vmi = libvmops.RunVMIAndExpectLaunch(vmi, 360)
+			Expect(console.LoginToAlpine(vmi)).To(Succeed())
+
+			By("Checking the guest ACPI table matches the one provided")
+			var batcher []expect.Batcher
+			if hasSlic {
+				batcher = append(batcher, []expect.Batcher{
 					&expect.BSnd{S: "xxd -p -c 40 /sys/firmware/acpi/tables/SLIC\n"},
-					&expect.BExp{R: console.RetValue(hexData)},
-				}, 3)).To(Succeed())
-			})
-		})
+					&expect.BExp{R: console.RetValue(bytesToHexStr(slicData))},
+				}...)
+			}
+			if hasMsdm {
+				batcher = append(batcher, []expect.Batcher{
+					&expect.BSnd{S: "xxd -p -c 40 /sys/firmware/acpi/tables/MSDM\n"},
+					&expect.BExp{R: console.RetValue(bytesToHexStr(msdmData))},
+				}...)
+			}
+			Expect(console.SafeExpectBatch(vmi, batcher, 10)).To(Succeed())
+		},
+			Entry("only slic table", 1, true, false),
+			Entry("only msdm table", 1, false, true),
+			Entry("slic and msdm tables", 2, true, true),
```

Comment:
Arguments to these entries are rather indirect, can you make it more explicit?

Reply:
You mean with a comment?